src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const onlyWomen = people.filter(person => person.sex === 'f');
+  const ages = onlyWomen.map(person => person.died - person.born);
+
+  const mothers = onlyWomen
+    .filter(person => people.find(child => child.mother === person.name))
+    .map(person => person.died - person.born);
```

Comment:
same here, you wrote redundant functionality for getting ages 2 times
let's define one general array at the start of the function depending on the `withChildren` argument

Reply:
I will edit it now, thx)